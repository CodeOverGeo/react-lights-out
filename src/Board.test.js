import React from 'react';
import { render } from '@testing-library/react';
import Board from './Board';

describe('<Board /> rendering', function () {
  describe('initial board, all lit, all unlit', function () {
    it('renders without crashing', function () {
      render(<Board />);
    });

    it('matches snapshot for a fully lit board', function () {
      const { asFragment } = render(<Board chanceLightStartsOn={1} />);
      expect(asFragment()).toMatchSnapshot();
    });
    it('matches snapshot for a fully unlit board', function () {
      const { asFragment } = render(<Board chanceLightStartsOn={0} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
