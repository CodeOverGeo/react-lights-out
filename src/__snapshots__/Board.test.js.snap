// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> rendering initial board, all lit, all unlit matches snapshot for a fully lit board 1`] = `
<DocumentFragment>
  <table
    class="Board"
  >
    <tbody>
      <tr>
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
        <td
          class="Cell Cell-lit"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`<Board /> rendering initial board, all lit, all unlit matches snapshot for a fully unlit board 1`] = `
<DocumentFragment>
  <div>
    You Win!
  </div>
</DocumentFragment>
`;
